### PR TITLE
[Feature] Allow consumer to pass in an IHttpClient in [#85]

### PR DESCRIPTION
**Pull request by chrisw000**
As per discussion on #85

### CODE REVIEW

**Comment by chrisw000** [comment]
BTW - on IHttpClient.... there is a typo on the method name...

SendA**S**ync should be SendAsync

I can rename and commit as part of this if you want...

File: GDAXSharp/HttpClient/IHttpClient.cs [hunk 1]
```diff
@@ -1,10 +1,13 @@
 ﻿using System.Net.Http;
+using System.Threading;
 using System.Threading.Tasks;
 
 namespace GDAXSharp.HttpClient
 {
     public interface IHttpClient
     {
+        Task<HttpResponseMessage> SendASync(HttpRequestMessage httpRequestMessage, CancellationToken cancellationToken);
```
Comment (dougdellolio) [inline GDAXSharp/HttpClient/IHttpClient.cs]: yes lets rename this :)

File: GDAXSharp.Specs/JsonFixtures/Reports/ReportResponseFixture.cs [hunk 2]
```diff
@@ -0,0 +1,27 @@
+﻿using GDAXSharp.Services.Reports.Models;
+
+namespace GDAXSharp.Specs.JsonFixtures.Reports
+{
+    public static class ReportResponseFixture
+    {
+        public static string Create(ReportType reportType)
```
Comment (dougdellolio) [inline GDAXSharp.Specs/JsonFixtures/Reports/ReportResponseFixture.cs]: why are these changes coming up?

**Comment by chrisw000** [comment]
It's quite possibly down to my lack of git experience.... and it's 02:30 AM :)

This is the same issue I had before... github thinks I'm still 1 commit behind the upstream/master ( the commit where you put in the report feature)

I've pulled upstream/master and merged/resolved any issues....

but it still thinks I'm 1 commit behind..... I'm obviously doing something wrong....

In the morning I'll trash my branch and do it again....  and create a new pull request....

**Comment by dougdellolio** [comment]
No worries. You may have to rebase against origin/master and then force push your branch back up. But either way works

**Comment by chrisw000** [comment]
OK - I did what you said.... force pushed the branch back up... that sorted it... thanks.

Different subject - the services all have private IHttpClient and IAuthenticator in the concrete classes, which are passed down to the methods at the abstract level.... but the abstract level also has private IHttpClient.... which references the same underlying object as the private IHttpClient in the concrete class - so it's unnecessary duplication.

Are you open to me cleaning this up - it removes quite a bit of clutter and clarifies whats going on - the Service constructors should just pass the IHttpClient and IAuthenticator directly to the base constructor

If so... I'll do a separate Pull Request tonight for that.

**Comment by dougdellolio** [comment]
@chrisw000 i will take care of it. thanks for pointing it out!

**Pull request merged**